crates/context_aware_config/src/api/default_config/handlers.rs

```diff
@@ -201,6 +172,118 @@ async fn create(
+    custom_headers: CustomHeaders,
+    request: web::Json<UpdateReq>,
+    db_conn: DbConnection,
+    #[cfg(feature = "high-performance-mode")] tenant: Tenant,
+    user: User,
+) -> superposition::Result<HttpResponse> {
+    let DbConnection(mut conn) = db_conn;
+    let req = request.into_inner();
+    let key_str = key.into_inner().into();
+    let tags = parse_config_tags(custom_headers.config_tags)?;
+
+    let existing = fetch_default_key(&key_str, &mut conn).map_err(|e| match e {
+        superposition::AppError::DbError(diesel::NotFound) => {
+            bad_argument!(
+                "No record found for {}. Use create endpoint instead.",
+                key_str
+            )
+        }
+        _ => {
+            log::error!("Failed to fetch {key_str}: {e}");
+            unexpected_error!("Something went wrong.")
+        }
+    })?;
+
+    let value = req.value.unwrap_or_else(|| existing.value.clone());
+    let schema = match req.schema {
+        Some(sch) => Value::Object(sch),
+        None => existing.schema.clone(),
+    };
+    let func_name = match req.function_name {
+        Some(Value::String(s)) => Some(s),
+        Some(Value::Null) => None,
+        None => existing.function_name.clone(),
+        Some(_) => return Err(bad_argument!("function_name must be string or null.")),
+    };
+
+    let updated_config = DefaultConfig {
+        key: key_str.to_owned(),
+        value,
+        schema,
+        function_name: func_name.clone(),
+        created_by: existing.created_by.clone(),
+        created_at: existing.created_at,
+        last_modified_at: Utc::now().naive_utc(),
+        last_modified_by: user.get_email(),
+    };
+
+    let jschema = JSONSchema::options()
+        .with_draft(Draft::Draft7)
+        .compile(&updated_config.schema)
+        .map_err(|e| {
+            log::info!("Failed to compile JSON schema: {e}");
+            bad_argument!("Invalid JSON schema.")
+        })?;
+
+    if let Err(e) = jschema.validate(&updated_config.value) {
+        let verrors = e.collect::<Vec<_>>();
+        log::info!("Validation failed: {:?}", verrors);
+        return Err(validation_error!(
+            "Schema validation failed: {}",
+            validation_err_to_str(verrors)
+                .get(0)
+                .unwrap_or(&String::new())
+        ));
+    }
+
+    if let Some(f_name) = &updated_config.function_name {
+        let function_code = get_published_function_code(&mut conn, f_name.to_string())
+            .map_err(|_| bad_argument!("Function {} doesn't exist.", f_name))?;
+        if let Some(f_code) = function_code {
+            validate_value_with_function(
+                f_name,
+                &f_code,
+                &updated_config.key,
+                &updated_config.value,
+            )?;
+        }
+    }
+
+    let version_id =
+        conn.transaction::<_, superposition::AppError, _>(|transaction_conn| {
+            diesel::insert_into(dsl::default_configs)
```

Comment:
lets use update instead of insert

Reply:
yeah its better to keep use update
but its cleaner like this if we use insert_into